contracts/Portfolio.sol

```diff
@@ -165,13 +165,20 @@ abstract contract PortfolioVirtual is Objective {
         // Checks
         if (to == address(this)) revert InvalidTransfer();
 
-        uint256 balance = getBalance(msg.sender, token);
-        if (amount == type(uint256).max) amount = balance;
-        if (amount > balance) revert DrawBalance();
+        uint8 decimals = IERC20(token).decimals();
+        uint256 amountWad = amount.scaleToWad(decimals);
+        uint256 balanceWad = getBalance(msg.sender, token);
+        if (amountWad == type(uint256).max) amountWad = balanceWad;
+        if (amountWad > balanceWad) revert DrawBalance();
 
         // Effects
-        _applyDebit(token, amount);
-        _decreaseReserves(token, amount);
+        _applyDebit(token, amountWad);
+        _decreaseReserves(token, amountWad);
+
+        // If we are using the full balance, we need to scale it from WAD.
+        if (amountWad == balanceWad) {
+            amount = amountWad.scaleFromWadDown(decimals);
+        }
```

Comment:
If you restructure as I suggest above, you can remove this extra conditional and do the `amount = amountWad.scaleFromWadDown(decimals)` in the first branch.

Reply:
(edited my comment above to do this)

Reply:
Decided to remove draw entirely in the pr #338